lua/autorun/client/cl_mcsp.lua

```diff
@@ -0,0 +1,38 @@
+local MCSPmodel = "models/milaco/minecraft_pm/minecraft_pm.mdl"
+
+hook.Add("PrePlayerDraw", "MCSPRenderSkin", function(ply) --Material ID 0 is the hat layer, ID 1 is the base layer
+	if IsValid(ply) and ply:GetModel() == MCSPmodel and ply:IsPlayer() then
+		render.MaterialOverrideByIndex(1, ImgurMaterial(ply:GetNWString("MCSPSkinURL", false), ply, ply:GetPos(), false, "VertexLitGeneric", {
+			["$alpha"] = 1,
+			["$halflambert"] = 1,
+			["$model"] = 1
+		}))
+		/*render.MaterialOverrideByIndex(0, ImgurMaterial(ply:GetNWString("MCSPSkinURL", false), ply, ply:GetPos(), false, "VertexLitGeneric", {
+			["$alphatest"] = 1,
+			["$halflambert"] = 1,
+			["$model"] = 1
+		}))*/ --Leave the hat layer out until we figure out what is causing the render issues
```

Comment:
what are the render issues? if you used to use $alpha for both layers there will be render issues where sometimes the base draws over the hat because $alpha doesn't z-write. it should be fine using $alphatest. either way I would prefer to have the hat layer in before releasing this

Reply:
There seems to be some sort of issue when setting the base layer to be fully opaque and the hat layer to use $alphatest. If I _only_ render the base layer, it works fine. If I render both the base and hat layer, it's almost like the transparent bits of the hat takes priority over the transparent bits of the base layer that were changed to be fully opaque. It's kind of hard to explain.
![20190825204525_1](https://user-images.githubusercontent.com/11602755/63663952-789c7300-c779-11e9-9753-a141f5588ee9.jpg)
![20190825204547_1](https://user-images.githubusercontent.com/11602755/63663964-7fc38100-c779-11e9-9f75-7bfb2cf084ba.jpg)
The first image is with just the base layer, and the second is both the base layer and hat layer.
I really can't figure out why it's causing this, I've tried changing render order, material parameters, etc. and nothing has fixed it thus far.

Reply:
I guess it should be noted that on this skin in particular, there is NOTHING on the hat layer. It's being used as a test to make sure the base layer stays opaque.

Reply:
are you sure it has the issue when using $alphatest (NOT $translucent)?

Reply:
Yes. I've always had the hat layer set to $alphatest. I've tried setting the base layer to $opaque 1 or $translucent 0 or whatever disables transparency on the material. I have tried using $translucent on the hat layer before just to see what would happen, and it did cause things to sort of render 'backwards' like you were talking about.

Reply:
Okay thanks, are you ready to merge?